rewrite-yaml/src/test/java/org/openrewrite/yaml/MergeYamlTest.java

```diff
@@ -1010,6 +1010,197 @@ void mergeSequenceMapChangeComplexMapping() {
+                  C:
+                    D:
+                      1: something else
+                      2: something else
+                      3: old desc
+                    D2:
+                      1: old description
+                    D3:
+                      1: old text
+              """,
+            """
+              A: # Some comment
+                B:
+                  C:
+                    D:
+                      1: something else
+                      2: something else
+                      3: old desc
+                      4: new desc
+                    D2:
+                      1: old description
+                      2: new description
+                    D3:
+                      1: old text
+                      2: new text
+                      3: more new text
+                  E: description
+              """
+          )
+        );
+    }
+
+    @Issue("https://github.com/openrewrite/rewrite/issues/2218")
+    @Test
+    void existingEntryBlockWithCommentAtLastLine() {
+        rewriteRun(
+          spec -> spec.recipe(new MergeYaml(
+            "$",
+            //language=yaml
+            """
+              spring:
+                application:
+                  description: a description
+              """,
+            false,
+            null,
+            null
+          )),
+          yaml(
+            """
+              spring:
+                application:
+                  name: main # some comment
+              """,
+            """
+              spring:
+                application:
+                  name: main # some comment
+                  description: a description
+              """
+          )
+        );
+    }
+
+    @Issue("https://github.com/openrewrite/rewrite/issues/2218")
+    @Test
+    void existingEntryBlockWithCommentAllOverThePlace() {
+        rewriteRun(
+          spec -> spec.recipe(new MergeYaml(
+            "$",
+            //language=yaml
+            """
+              A:
+                B:
+                  C:
+                    D:
+                      3: new desc
+                    D2:
+                      4: d
+                    D3:
+                      2: new description
+                    D4:
+                      2: new text
+              """,
+            false,
+            null,
+            null
+          )),
+          yaml(
+            """
+              A: # Some comment
+                B: # Some comment 2
+                  C: # Some comment 3
+                    D: # Some comment 4
+                      1: something else
+                      2: old desc # Some comment 5
+                    D2:
+                      1: a
+                      2: b # Some comment 10
+                      3: c
+                    D3: # Some comment 6
+                      1: old description # Some comment 7
+                    D4: # Some comment 8
+                      1: old text # Some comment 9
```

Comment:
Right now there's a mix of comments I suspect are not even visited, but they are indistinguishable from comments that are relevant to the recipe. I feel there's a bit of a missed opportunity here in the comments to clarify what the recipe should be doing. How about instead of `Some comment` we describe what we expect for that line & comment with comments like
- `Comment untouched`,
- `Comment moved from next block to previous element`